File: DesignPatternsDotNet/CommandPattern/Commands/VolumeUp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CommandPattern.Commands
{
   public class VolumeUp : ICommand{
	private IReceiver myReceiver  = null;
	public VolumeUp(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void Execute() {
		myReceiver.VolumeUp();;	
	}
	public void Undo() {
		myReceiver.VolumeDowne();;
	}
}
}

File: DesignPatternsJava/src/com/command/VolumeUp.java
package com.command;

public class VolumeUp implements ICommand{
	private IReceiver myReceiver  = null;
	public VolumeUp(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void execute() {
		myReceiver.volumeUp();;	
	}
	public void undo() {
		myReceiver.volumeDowne();;
	}
}
